docs/enhancements/infrastructure-monitoring-service-api/alarms.md

```diff
@@ -0,0 +1,669 @@
+2. Response with retrieved instance of `probableCause` and appropriate code
+
+
+### `internal` family
+
+NOTE: These APIs are not exposed to client and available only internally 
+
+#### Steps `internal/v1/caas-alerts/alertmanager`
+This API is activated only after configuring ACM alertmanager that can it can call back alarm service
+
+A minimal configuration
+```yaml
+route:
+  receiver: webhook_receiver
+receivers:
+  - name: webhook_receiver
+    webhook_configs:
+      - url: "o-ran-inventory-api-alarms.kubernetes.svc/internal/v1/caasAlerts/alertmanager"
+        send_resolved: true 
+```
+
+```shell
+oc -n open-cluster-management-observability create secret generic alertmanager-config --from-file=alertmanager.yaml --dry-run -o=yaml |  oc -n open-cluster-management-observability replace secret --filename=-
+```
+
+1. Example payload
+    ```json
+   "receiver": "webhook_receiver",
+   "status": "firing"
+   "alerts": [
+        {
+            "status": "firing",
+            "labels": {
+                "alertname": "UpdateAvailable",
+                "channel": "stable-4.16",
+                "managed_cluster": "89070983-a62f-4dbe-9457-7e0c27832c63",
+                "namespace": "openshift-cluster-version",
+                "openshift_io_alert_source": "platform",
+                "prometheus": "openshift-monitoring/k8s",
+                "severity": "info",
+                "upstream": "<default>"
+            },
+            "annotations": {
+                "description": "For more information refer to 'oc adm upgrade' or https://console-openshift-console.apps.cnfdf27.sno.telco5gran.eng.rdu2.redhat.com/settings/cluster/.",
+                "summary": "Your upstream update recommendation service recommends you update your cluster."
+            },
+            "startsAt": "2024-08-28T11:39:17.958Z",
+            "endsAt": "0001-01-01T00:00:00Z",
+            "generatorURL": "https://console-openshift-console.apps.cnfdf27.sno.telco5gran.eng.rdu2.redhat.com/monitoring/graph?g0.expr=sum+by+%28channel%2C+namespace%2C+upstream%29+%28cluster_version_available_updates%29+%3E+0&g0.tab=1",
+            "fingerprint": "91406cd113ad87e5"
+        },
+   ]
+    ```
+2. Sync `alarm_event_record` Table
+   - Update rows to "resolved" that are missing in the current payload (i.e previously seen but somehow missed the "resolved" notification)
+     ```sql
+      UPDATE alarm_event_record
+      SET status = 'resolved', alarm_cleared_time = CURRENT_TIMESTAMP
+      WHERE (finger_print, alarm_raised_time) NOT IN (
+             ('Something-Thats-Only-Now-Available', '2023-09-01 10:02:00+00')
+      );
+     ```
+   - Create new AlarmEventRecord. Alert entry payload will require Alarm Definition ID and Probable Cause ID which can retrieved as shown [here](#for-a-given-resourcetypeid-and-alarmname-coming-from-am-alert-find-the-alarmdefinitionid-and-probablecauseid)
```

Comment:
nit: the "here" link doesn't work for me - maybe just an issue with the preview in github?